components/butterfly/tests/encryption/mod.rs

```diff
@@ -6,7 +6,7 @@ use habitat_core::crypto::keys::sym_key::SymKey;
 fn symmetric_encryption_of_wire_payloads() {
     let ring_key = SymKey::generate_pair_for_ring("wolverine").expect("Failed to generate an in \
                                                                        memory symkey");
-    let mut net = btest::SwimNet::new_ring_encryption(2, &ring_key);
+    let mut net = btest::SwimNet::new_ring_encryption_rhw(2, &ring_key);
```

Comment:
Does `new_ring_encryption` (and `new`, for that matter) need locking annotations? It's a constructor, so it has full control over all the data structures in question and nobody else can access them until the function returns.

Reply:
Mostly I'm just trying to be consistent with all this, rather than following an unwritten checklist in my head for when to add annotations and when not to.

Reply:
Constructors in rust don't have any special treatment as far as the language goes, so they should be treated like regular functions. The point of the locking annotations is to make it clear what locks will be acquired in which code paths. We don't want code that already holds a `RumorHeat` lock to call `new` without realizing it also takes a lock because that could result in deadlock.

That said, I haven't been bothering with the annotations in test-only code, so I could really do either way in this particular case.